File: 929.unique-email-addresses.js
/*
 * @lc app=leetcode id=929 lang=javascript
 *
 * [929] Unique Email Addresses
 */

// @lc code=start
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function(emails) {
  let validEmails = {}
  emails.forEach(email=>{
    if (/[^a-z.+@]/.test(email)) return
    if ((email = email.split('@')).length > 2) return

    email[0] = email[0].split('+')[0]
    email = email[0].split('.').join('') + '@' + email[1]
    validEmails[email] = true
  })
  return Object.keys(validEmails).length
};
// @lc code=end

File: 929.unique-email-addresses.py
#
# @lc app=leetcode id=929 lang=python3
#
# [929] Unique Email Addresses
#

# @lc code=start
import re
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        result = {}
        for email in emails:
            if (re.search("[^a-zA-Z.+]", email) is None): continue
            email = email.split('@')
            if (len(email) != 2): continue

            email[0] = email[0].split('+')[0].replace('.','')
            result['@'.join(email).lower()] = True

        return len(result)
# @lc code=end

